src/common/utils/log/logger.go

```diff
@@ -79,6 +88,58 @@ func DefaultLogger() *Logger {
 	return logger
 }
 
+func (l *Logger) clone() *Logger {
+	return &Logger{
+		out:       l.out,
+		fmtter:    l.fmtter,
+		lvl:       l.lvl,
+		callDepth: l.callDepth,
+		skipLine:  l.skipLine,
+		fields:    l.fields,
+		fieldsStr: l.fieldsStr,
+		mu:        l.mu,
+	}
+}
+
+// WithDepth ...
+func (l *Logger) WithDepth(depth int) *Logger {
+	r := l.clone()
+	r.callDepth = depth
+
+	return r
+}
+
+// WithFields ...
+func (l *Logger) WithFields(fields Fields) *Logger {
```

Comment:
Can we make Withxx and Setxxx consistent?  Or protect Setxxx from being called by mistake?

Reply:
SetXxx changed to private method to make logger is immutable.